pypesto/ensemble/utils.py

```diff
@@ -230,3 +237,57 @@ def get_prediction_dataset(ens: Union[Ensemble, EnsemblePrediction],
                         'an EnsemblePrediction object as input. Stopping.')
 
     return dataset
+
+
+def read_ensemble_prediction_from_h5(
+        predictor: Union[Callable[[Sequence], PredictionResult], None],
+        input_file: str):
+
+    # open file
+    with h5py.File(input_file, 'r') as f:
+        pred_res_list = []
+        bounds = {}
+        for key in f.keys():
+            if key == 'prediction_id':
+                prediction_id = f[key][()].decode()
+                continue
+            if key in {'lower_bounds', 'upper_bounds'}:
+                if isinstance(f[key], h5py._hl.dataset.Dataset):
+                    bounds[key] = f[key][:]
+                    continue
+                bounds[key] = [f[f'{key}/{cond}'][()]
+                               for cond in f[key].keys()]
+                bounds[key] = np.array(bounds[key])
+                continue
+            x_names = decode_array(f[f'{key}/x_names'][()])
+            condition_ids = np.array(decode_array(
+                f[f'{key}/condition_ids'][()]
+            ))
+            pred_cond_res_list = []
+            for id, _ in enumerate(condition_ids):
+                output = f[f'{key}/{id}/output'][:]
+                output_ids = decode_array(f[f'{key}/{id}/output_ids'][:])
+                timepoints = f[f'{key}/{id}/timepoints'][:]
+                pred_cond_res_list.append(PredictionConditionResult(
+                    timepoints=timepoints,
+                    output_ids=output_ids,
+                    output=output,
+                    x_names=x_names
+                ))
+            pred_res_list.append(PredictionResult(
+                conditions=pred_cond_res_list,
+                condition_ids=condition_ids
+            ))
+        return EnsemblePrediction(predictor=predictor,
+                                  prediction_id=prediction_id,
+                                  prediction_results=pred_res_list,
+                                  lower_bound=bounds['lower_bounds'],
+                                  upper_bound=bounds['upper_bounds'],
+                                  )
+
+
+def decode_array(array: np.ndarray) -> np.ndarray:
+    """Decodes array of bytes to string"""
+    for i in range(len(array)):
+        array[i] = array[i].decode()
+    return array
```

Comment:
Seems inefficient, is `array.astype(str)` equivalent?

Reply:
I think (will test it again) in those cases, the array would be `b"str"` and I wanted to get rid of the `b` in front

Reply:
Perhaps https://numpy.org/doc/stable/reference/generated/numpy.char.decode.html ?

Reply:
this does not work. probably since the type of the array is `object` and not `str` I guess.